common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/CurrentTaskAuthorizationActivity.java

```diff
@@ -0,0 +1,163 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.providers.oauth2;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.localbroadcastmanager.content.LocalBroadcastManager;
+
+import com.microsoft.identity.common.internal.ui.DualScreenActivity;
+import com.microsoft.identity.common.logging.Logger;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentAction.DESTROY_REDIRECT_RECEIVING_ACTIVITY_ACTION;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentAction.REDIRECT_RETURNED_ACTION;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentAction.REFRESH_TO_CLOSE;
+
+/**
+ * Authorization activity for addressing authorization activities that are launched within the task
+ * associated with the activity provided as a parameter to InteractiveTokenCommand
+ *
+ * NOTE: this is only used when library configuration (set in MSAL) indicates that it should be used.
+ * DEFAULT today is to create a new task for authorization; however this leads to problems if the user
+ * navigates away during authorization (to home screen for example) or when using multi-window.
+ */
+public class CurrentTaskAuthorizationActivity extends DualScreenActivity {
+
+    private static final String TAG = CurrentTaskAuthorizationActivity.class.getSimpleName();
+
+    private CurrentTaskBrowserAuthorizationFragment mFragment;
+    private boolean mCloseCustomTabs = true;
+    private BroadcastReceiver redirectReceiver;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        final Fragment fragment = AuthorizationActivityFactory.getAuthorizationFragmentFromStartIntent(getIntent());
+
+        if(fragment instanceof CurrentTaskBrowserAuthorizationFragment){
+            mFragment = (CurrentTaskBrowserAuthorizationFragment) fragment;
+            mFragment.setInstanceState(getIntent().getExtras());
+        }else{
+            IllegalStateException ex = new IllegalStateException("Unexpected fragment type");
+            Logger.error(TAG, "Fragment provided was not of type CurrentTaskBrowserAuthorizationFragment", ex);
+            throw ex;
+        }
+
+        //This will be invoked when CurrentTaskAuthorizationActivity is not found in the current task stack... hence it will create fresh
+        if (REDIRECT_RETURNED_ACTION.equals(getIntent().getAction())) {
+            if(CurrentTaskBrowserAuthorizationFragment.class.isInstance(mFragment)) {
+                Bundle arguments = new Bundle();
+                arguments.putBoolean("RESPONSE", true);
```

Comment:
"RESPONSE" can be defined in a constant somewhere

Reply:
Yep.